proto/client-api/src/lib.rs

```diff
@@ -113,11 +113,75 @@ pub enum ServerMsg {
 
     /// `Media Server` notifies `Client` about happened facts and it reacts on
     /// them to reach the proper state.
-    Event(Event),
+    Event {
+        /// ID of `Room` that this [`Event`] is associated with.
+        room_id: RoomId,
+
+        /// Actual [`Event`] sent to `Client`.
+        event: Event,
+    },
 
     /// `Media Server` notifies `Client` about necessity to update its RPC
     /// settings.
     RpcSettings(RpcSettings),
+
+    /// `Media Server` notifies `Client` that he joined `Room`.
+    JoinedRoom {
+        /// ID of `Room` into which `Client` was joined.
+        room_id: RoomId,
+
+        /// ID of `Member` which joined `Room`.
+        member_id: MemberId,
```

Comment:
@evdokimovs @alexlapa why is this a separate type of `ServerMsg`? It looks like an `Event`.

Reply:
@tyranron ,

Разделение зон ответственности. За `Event`'ы отвечает `Room`а, за `ServerMsg::RpcSettings` отвечает `WebSocketRpcClient`, а за авторизацию `WebSocketRpcSession`. `Room`е не нужно знать о том, что пришел `ServerMsg::JoinedRoom`, поскольку она с этим ничего не должна делать, поэтому удобнее выделять это в отдельный `ServerMsg`.

Reply:
@tyranron ,

`JoinedRoom` и `LeftRoom` это session-менеджмент: авторизация и завершение сессии, это ближе к транспорту чем к медиа. 

В таком виде все достаточно удобное склеивается. Путь сообщения от сервера к комнате на клиенте таков: `WebSocketRpcClient` => `RpcSession` => `Room`. 
1. `WebSocketRpcClient` обрабатывает `RpcSettings` и `Ping`/`Pong`'и, а все остальное передает дальше. 
2. `RpcSession` обрабатывает `JoinedRoom` и `LeftRoom`, а все остальное передает дальше. 
3. А вот `Room` уже работает исключительно с ивенатми.

Reply:
@alexlapa @evdokimovs from the description and its contents they really look like events. And symetrical client messages look like commands 🤷‍♂️

Given the condition that the `Room` is only one of possible media elements supported by media server, and responsible for the authorization, I don't see any reasons why these messages deserve a separate notion in protocol. What if media server pipeline has no authorization or `Room`?

`JoinedRoom`/`LeftRoom` should really be `Room`-related events, and `JoinRoom`/`LeftRoom` should really be `Room`-related commands. This, however, doesn't conflict with the concerns you've decribed above:

> Разделение зон ответственности. За `Event`'ы отвечает `Room`а, за `ServerMsg::RpcSettings` отвечает `WebSocketRpcClient`, а за авторизацию `WebSocketRpcSession`. `Room`е не нужно знать о том, что пришел `ServerMsg::JoinedRoom`, поскольку она с этим ничего не должна делать, поэтому удобнее выделять это в отдельный `ServerMsg`.

That's implementation defined. The protocol itself, as an interface, should not depend on how it's implemented on server or client side. Someday it may change, so we should change the protocol too? Nope.

> В таком виде все достаточно удобное склеивается. Путь сообщения от сервера к комнате на клиенте таков: `WebSocketRpcClient` => `RpcSession` => `Room`.

Okay. Again, that's the implementation only, not the protocol itself.

> 2. `RpcSession` обрабатывает `JoinedRoom` и `LeftRoom`, а все остальное передает дальше. 
> 3. А вот `Room` уже работает исключительно с ивенатми.

Given that `RpcSession` sits between `WebSocketRpcClient` and the `Room`, there is no problem for it to intercept `Event::JoinedRoom`/`Event::LeftRoom` the same way. There is no conflict. Yet, we keep the protocol cleaner itself.